exchanges/huobi/huobi_test.go

```diff
@@ -2870,74 +2851,52 @@ func TestUpdateTickers(t *testing.T) {
-	if assert.NoError(t, err) {
-		assert.NotZero(t, tick.Close)
-	}
-
-	cp = currency.NewPair(currency.BTC, currency.NewCode("CQ"))
-	cp, err = h.convertContractShortHandToExpiry(cp, tt)
-	assert.NoError(t, err)
-	assert.NotEqual(t, "CQ", cp.Quote.String())
-	tick, err = h.FetchTicker(context.Background(), cp, asset.Futures)
-	if assert.NoError(t, err) {
-		assert.NotZero(t, tick.Close)
-	}
-
-	// calculate a specific date
-	cp = currency.NewPair(currency.BTC, currency.NewCode("CQ"))
-	tt = time.Date(2021, 6, 3, 0, 0, 0, 0, time.UTC)
-	cp, err = h.convertContractShortHandToExpiry(cp, tt)
-	assert.NoError(t, err)
-	assert.Equal(t, "210625", cp.Quote.String())
 
-	cp = currency.NewPair(currency.BTC, currency.NewCode("CW"))
-	cp, err = h.convertContractShortHandToExpiry(cp, tt)
-	assert.NoError(t, err)
-	assert.Equal(t, "210604", cp.Quote.String())
+	h := new(HUOBI) //nolint:govet // Intentional shadow to avoid future copy/paste mistakes
+	require.NoError(t, testexch.Setup(h), "Test Instance Setup must not fail")
 
-	cp = currency.NewPair(currency.BTC, currency.NewCode("CWif hat"))
-	_, err = h.convertContractShortHandToExpiry(cp, tt)
-	assert.ErrorIs(t, err, errInvalidContractType)
+	_, err := h.FetchTradablePairs(context.Background(), asset.Futures)
+	require.NoError(t, err)
 
-	tt = time.Now()
-	cp = currency.NewPair(currency.BTC, currency.NewCode("NQ"))
-	cp, err = h.convertContractShortHandToExpiry(cp, tt)
-	assert.NoError(t, err)
-	assert.NotEqual(t, "NQ", cp.Quote.String())
-	tick, err = h.FetchTicker(context.Background(), cp, asset.Futures)
-	if err != nil {
-		// Huobi doesn't always have a next-quarter contract, return if no data found
-		return
+	n := time.Now().Truncate(24 * time.Hour)
+	for _, cType := range contractExpiryNames {
+		p, err := h.pairFromContractExpiryCode(currency.Pair{
+			Base:  currency.BTC,
+			Quote: currency.NewCode(cType),
+		})
+		if cType == "NQ" && err != nil {
+			continue // Next Quarter is intermittently present
+		}
+		require.NoErrorf(t, err, "pairFromContractExpiryCode must not error for %s code", cType)
+		assert.Equal(t, currency.BTC, p.Base, "pair Base should be the same")
+		h.futureContractCodesMutex.RLock()
+		exp, ok := h.futureContractCodes[cType]
+		h.futureContractCodesMutex.RUnlock()
+		require.True(t, ok, "%s type must be in contractExpiryNames", cType)
+		assert.Equal(t, currency.BTC, p.Base, "pair Base should be the same")
+		assert.Equal(t, exp, p.Quote, "pair Quote should be the same")
+		d, err := time.Parse("060102", p.Quote.String())
+		require.NoError(t, err, "currency code must be a parsable date")
+		require.Truef(t, d.After(n), "%s expiry must be today or after", cType)
+		switch cType {
+		case "CW", "NW":
+			require.True(t, d.Before(n.Add(24*time.Hour*14)), "%s expiry must be within 2 weeks", cType)
+		case "CQ", "NQ":
+			require.True(t, d.Before(n.Add(24*time.Hour*90*2)), "%s expiry must be within 2 months", cType)
```

Comment:
Doesnt sound right for Next Quarter if its ever present. It would likely be further out. But then I also take issue with 180 days being within 2 months

Reply:
Agreed. 2 quarters.

Fixed 8110c76d